### PR TITLE
librum: init at 0.12.1

### LINKED ISSUE #256018: Package request: Librum
**Project description**

Librum is an e-book reader that can download books from a personal online library.

**Metadata**

* homepage URL: https://librumreader.com/
* source URL: https://github.com/Librum-Reader/Librum
* license: gpl3
* platforms: gnu/linux, macos, windows

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/256018

@oluceps agreed to be one of maintainers.

image

image

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by oluceps** [comment]
Result of `nixpkgs-review pr 293031` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>librum</li>
  </ul>
</details>

**Review by oluceps — APPROVED** [review]

**Review by oluceps — APPROVED** [review]

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 293031` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>librum</li>
  </ul>
</details>

**Comment by Aleksanaa** [comment]
> Result of `nixpkgs-review pr 293031` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 1 package built:
> 
>     * librum

Build on Darwin is fixed?

**Comment by sikmir** [comment]
> > Result of `nixpkgs-review pr 293031` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> > 1 package built:
> > ```
> > * librum
> > ```
> 
> Build on Darwin is fixed?

Local build is OK, but CI still failed.

**Comment by Aleksanaa** [comment]
> Local build is OK, but CI still failed.

That's caused by a dependency failure which should be fixed later. So no problem I think. Thanks for testing.

File: pkgs/by-name/li/librum/package.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      nixMupdfLibPath = "${mupdf-cxx.out}/lib";
+      nixMupdfIncludePath = "${mupdf-cxx.dev}/include";
+    })
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    qt6.qttools
+    qt6.wrapQtAppsHook
+  ];
```
Comment (sikmir) [inline pkgs/by-name/li/librum/package.nix:37]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [ desktopToDarwinBundle ];
```

Comment (Aleksanaa) [reply pkgs/by-name/li/librum/package.nix:37]: Wait, Darwin doesn’t recognize .desktop files?

**Review by sikmir — APPROVED** [review]

File: pkgs/by-name/li/librum/package.nix (hunk 1 above)
Comment (sikmir) [reply pkgs/by-name/li/librum/package.nix:37]: > Wait, Darwin doesn’t recognize .desktop files?

No)

Comment (sikmir) [reply pkgs/by-name/li/librum/package.nix:37]: Result is:
```
$ tree result
result
├── Applications
│   └── Librum.app
│       └── Contents
│           ├── Info.plist
│           ├── MacOS
│           │   └── Librum -> ../../../../bin/librum
│           └── Resources
│               └── librum.icns
├── bin
│   └── librum
├── lib
│   ├── libadapters.dylib
│   ├── libapplication.dylib
│   ├── libdomain.dylib
│   ├── libinfrastructure.dylib
│   └── libpresentation.dylib
└── share
    ├── applications
    │   └── librum.desktop
    └── pixmaps
        └── librum.svg
```

Comment (Aleksanaa) [reply pkgs/by-name/li/librum/package.nix:37]: Is this the only way this application will appear on the launcher page? (If so, is nixpkgs missing tons of `desktopToDarwinBundle`)

Comment (sikmir) [reply pkgs/by-name/li/librum/package.nix:37]: Many Qt-based apps produce proper *.app bundle as build result, so no need `desktopToDarwinBundle`. But yes, there is still many GUI apps where `desktopToDarwinBundle` would be good to add.

File: pkgs/by-name/li/librum/package.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  buildInputs = [
+    qt6.qtbase
+    qt6.qtsvg
+  ] ++ lib.optionals stdenv.isLinux [
+    qt6.qtwayland
+  ] ++ lib.optionals stdenv.isDarwin [
+    desktopToDarwinBundle
```
Comment (sikmir) [inline pkgs/by-name/li/librum/package.nix:46]: It's nativeBuildInputs.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 293031` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>librum</li>
  </ul>
</details

Tried to launch and use, LGTM.

**Pull request merged**